File: 257-binary-tree-paths/257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<string> binaryTreePaths(TreeNode* root)
    {
        vector<string> result;
        //Edge case
        if(!root)
            return result;
        
        binaryPaths(result,root,to_string(root -> val));
        return result;
    }
    void binaryPaths(vector<string> &result,TreeNode* root,string t)
    {
        if(!root -> left && !root -> right){
            result.push_back(t);
            return;
        }
        if(root -> left)
            binaryPaths(result,root -> left,t + "->" + to_string(root -> left -> val));
        if(root -> right)
            binaryPaths(result,root -> right,t + "->" + to_string(root -> right -> val));  
    }
};

File: 257-binary-tree-paths/257-binary-tree-paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    let result = [];
    if(!root)
        return result;
    binaryPaths(result,root,root.val.toString())
    return result;
};

let binaryPaths = function(result,root,t){
    if(!root.left && !root.right){
        result.push(t);
        return;
    }
    if(root.left)
        binaryPaths(result,root.left,t + "->" + root.left.val.toString());
    if(root.right)
        binaryPaths(result,root.right,t + "->" + root.right.val.toString());
};